perf: Change return type of custom protocol handler from `Vec<u8>` to `Cow<'static, [u8]>`, closes #796

rhysd opened the pull request:
<!--
Update "[ ]" to "[x]" to check a box

Please make sure to read the Pull Request Guidelines: https://github.com/tauri-apps/tauri/blob/dev/.github/CONTRIBUTING.md#pull-request-guidelines
-->

### What kind of change does this PR introduce?
<!-- Check at least one. If you are introducing a new binding, you must reference an issue where this binding has been proposed, discussed and approved by the maintainers. -->

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?
<!-- If yes, please describe the impact and migration path for existing applications in an attached issue. -->

- [x] Yes, and the changes were approved in issue #796
- [ ] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/wry/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### Other information

I confirmed this change using [the custom protocol example](https://github.com/tauri-apps/wry/blob/dev/examples/custom_protocol.rs) on the following platforms:

- Windows 10
- macOS 11.7
- Ubuntu 22.04

I don't have Android and iOS development environment so I could not confirm this patch on it.

rhysd committed:
perf: Change return type of custom protocol handler from `Vec<u8>` to `Cow<'static, [u8]>`

to allow returning static resources without heap allocation

amrbashir commented on examples/custom_protocol.rs:
```diff
@@ -27,21 +29,19 @@ fn main() -> wry::Result<()> {
+      let content = if path == "/" {
+        PAGE1_HTML.into()
+      } else {
+        // 1.. for removing leading slash
```
```suggestion
        // `1..` for removing leading slash
```

amrbashir commented on examples/menu.rs:
```diff
@@ -43,21 +45,19 @@ fn main() -> wry::Result<()> {
+      let content = if path == "/" {
+        PAGE1_HTML.into()
+      } else {
+        // 1.. for removing leading slash
```
```suggestion
        // `1..` for removing leading slash
```

amrbashir reviewed (dismissed):
Thank you! LGTM! regardless of the few nitpicks.

amrbashir commented:
for android, you can just install `rustup target add aarch64-linux-android` and then run `cargo check --target aarch64-linux-android` to make sure it compiles fine.

rhysd commented:
> for android, you can just install rustup target add aarch64-linux-android and then run cargo check --target aarch64-linux-android to make sure it compiles fine.

Ah, I didn't know that checking non-host target is so easy with cargo. Thank you. I'll check the compile errors and your review points tonight.

rhysd committed:
fix(android): Change return type of response to `Cow`

rhysd committed:
chore(examples): Fix some comments

following the review https://github.com/tauri-apps/wry/pull/797#pullrequestreview-1210580935

rhysd commented:
@amrbashir @wusyong I fixed Android build and resolved review comments. I apologize that I could not take time to work on this yesterday.

amrbashir reviewed (approved):
LGTM! Thank you.
